### ISSUE TITLE
Use CloudEvents SDK

### ISSUE THREAD

**Issue by grant** [GoogleCloudPlatform/functions-framework-nodejs#140]
Right now we don't use a CloudEvents SDK.

That's mostly because there isn't a usable one right now.

Let's fix that. (I'm working on it.)

---

UPDATE July 3: The JavaScript SDK is going to support TypeScript, so we'll eventually use that.

**Comment by grant** [comment]
https://github.com/cloudevents/sdk-typescript/pull/1

**Comment by grant** [comment]
Aim to re-check in September.

**Comment by grant** [comment]
There aren't many benefits for using the CE SDK at this time rather than our own implementation. The SDK is also rapidly changing. Going to close this issue.

**Comment by lance** [comment]
@grant I was wondering if you have considered revisiting this. I think the community would benefit from Google's use of the project, and it seemed for a while that you were invested in the community. The churn from the conversion to TypeScript has settled, so major version releases are less frequent. I may have misunderstood your previous intent, but I thought your push to convert the project to TypeScript was for exactly this reason - so that you could use the community / CNCF library. It's disappointing to me that after all of that effort, it seems GCF is walking away from the project anyway.

**Comment by grant** [comment]
Hey @lance! Good to hear from you.

Yes, we should integrate the Node CESDK into this repo. We've integrated the SDK for projects with python, ruby, .NET, Java, PHP already. GCF and Google Cloud is definitely not walking away from CloudEvents. We're investing more here in the future. Really glad where we landed with the Node CESDK!

For Functions, a lot of the impact of CloudEvents in certain languages won't land until next year. There hasn't been a rush here and internal priorities change all the time.

We should revisit this. I created: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/391
